### PR TITLE
DCOS-40469: update http service

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
## Testing

1. Install new module `$ npm i`
2. Check out the corresponding plugin branch
3. Check the UI and see if this change introduced a bug in the areas touched by the update.

## Trade-offs

- Decided to keep the MetronomeClient open, so that it consumers can decide if they want to get the status code

## Dependencies

None

## Screenshots

No visual change

### CODE REVIEW

**Comment by nLight** [comment]
Tests are failing

**Comment by nLight** [comment]
Tasks keep loading
screen shot 2018-08-28 at 15 53 58

**Comment by DanielMSchmidt** [comment]
@nLight I don't see why, I changed nothing in that direction, at least if I look into the files.

File: src/js/stores/MesosStateStore.js [hunk 1]
```diff
@@ -81,10 +81,11 @@ class MesosStateStore extends GetSetBaseStore {
       "/mesos/api/v1?get_master"
     )
       .retryWhen(linearBackoff(RETRY_DELAY, MAX_RETRIES))
-      .do(message => {
+      .map(({ response }) => response)
```
Comment (nLight) [inline src/js/stores/MesosStateStore.js]: `@dcos/mesos-client` still depends on `@dcos/http-service@0.3.0` maybe that's the issue?

Comment (DanielMSchmidt) [reply src/js/stores/MesosStateStore.js]: But this is `getMasterRequest` which should be a normal `request`, shouldn't it? Do you think two different versions are a problem?

Comment (nLight) [reply src/js/stores/MesosStateStore.js]: https://github.com/dcos/dcos-ui/pull/3216/files#diff-22eec13077e445e982cd1a9232667028R2

Comment (DanielMSchmidt) [reply src/js/stores/MesosStateStore.js]: Ah got it, it's also called request but from some other library 🙈

**Comment by nLight** [comment]
Gotta bump `@dcos/mesos-client@0.2.5` as well

**Comment by natmegs** [comment]
Noticed a couple areas where it looks like there's issues loading data. (Maybe I am missing a dependency? Do I have to have anything else set up outside of this branch/repo?). I'll look in to it.

Node Agents > Detail (Not loading)
node-detail

Cluster Overview (Mesos details not loading)
cluster-overview

Catalog (icons not loading)
catalog

**Comment by nLight** [comment]
@DanielMSchmidt flicts

**Comment by DanielMSchmidt** [comment]
System tests can't start the cluster, the other tests were green. I am retriggering now

File: src/typings/dcos__http-service.d.ts [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
+  code: number;
+  message: string;
+  response: T;
+}
 
+export function request<T>(
+  url: string,
+  options?: RequestOptions
+): Observable<RequestResponse<T>>;
+export function stream<T>(
```
Comment (nLight) [inline src/typings/dcos__http-service.d.ts]: But does stream return `RequestResponse`?

Comment (DanielMSchmidt) [reply src/typings/dcos__http-service.d.ts]: No it doesn't, fixed the typo

**Comment by DanielMSchmidt** [comment]
@nLight Fixed the typo, please take another look!

**Comment by nLight** [comment]
System tests are broken :( and I don't think that is your fault

**Comment by DanielMSchmidt** [comment]
CI is gree @dcos/frontend Let's get this done 👍

**Pull request merged**